wagtail/search/tests/elasticsearch_common_tests.py

```diff
@@ -176,6 +176,12 @@ def test_slice_to_next_page(self):
         results = self.backend.search(MATCH_ALL, models.Book)[110:]
         self.assertEqual(len(results), 54)
 
+    def test_search_with_date_filter(self):
+        after_1900 = models.Book.objects.filter(publication_date__year__gt=1900)
```

Comment:
Could you add tests for months/days as well?

Reply:
Turns out this doesn't work for months/days, I'm going to keep looking into it